src/main/java/org/medicmobile/webapp/mobile/ChtExternalApp.java

```diff
@@ -260,33 +265,36 @@ private JSONObject parseBundleToJson(Bundle bundle) {
 			return null;
 		}
 
-		private void setValueInJson(String key, JSONObject json, Bundle bundle) {
+		private void setValueInJson(String parentKey, String childKey, JSONObject json, Bundle bundle) {
+			String key = ("detectedRdt".equals(parentKey) && "concerns".equals(childKey))
+				? "detectionConcerns"
+				: childKey;
 			try {
-				Object value = bundle.get(key);
+				Object value = bundle.get(childKey);
 
 				if (value instanceof Bitmap) {
-					json.put(key, parseBitmapImageToBase64((Bitmap) value));
+					json.put(key, parseBitmapImageToBase64((Bitmap) value, false));
 					return;
 				}
 
 				if (value instanceof Bundle) {
-					json.put(key, parseBundleToJson((Bundle) value));
+					json.put(key, parseBundleToJson(key, (Bundle) value));
 					return;
 				}
 
 				if (isBundleList(value)) {
 					JSONArray jsonArray = ((List<Bundle>) value)
 							.stream()
-							.map(this::parseBundleToJson)
+							.map(val -> this.parseBundleToJson(key, val))
 							.collect(Collector.of(JSONArray::new, JSONArray::put, JSONArray::put));
-
 					json.put(key, jsonArray);
 					return;
 				}
 
 				Optional<Uri> imagePath = getImageUri(value);
 				if (imagePath.isPresent()) {
-					json.put(key, getImageFromStoragePath(imagePath.get()));
+					boolean keepFullResolution = bundle.getBoolean("sampleImage", false);
+					json.put(key, getImageFromStoragePath(imagePath.get(), keepFullResolution));
```

Comment:
This tweak allows us to save the returned image uncompressed when `samepleImage` is true. This is needed to be able to return the images marked for sampling to Audere (they cannot use compressed versions of the images when training the model).  

Note that the difference between `100` quality (full resolution) and `75` quality (compressed) may seem minor, but in my unscientific measurements I was seeing the ultimate size of the `png` file for the full resolution image to be `~5x` greater than the compressed image.  For example, when testing with my phone with a `16MP` camera, the compressed image size (of the png attachment uploaded to Couch) was `1.4MB`, but the uncompressed image size (saved when `sampleImage: true`) was `5.8MB`.